openedx/features/enterprise_support/api.py

```diff
@@ -327,14 +322,79 @@ def enterprise_enabled():
     return 'enterprise' in settings.INSTALLED_APPS and settings.FEATURES.get('ENABLE_ENTERPRISE_INTEGRATION', False)
 
 
+def enterprise_is_enabled(otherwise=None):
+    """Decorator which requires that the Enterprise feature be enabled before the function can run."""
+    def decorator(func):
+        """Decorator for ensuring the Enterprise feature is enabled."""
+        def wrapper(*args, **kwargs):
+            if enterprise_enabled():
+                return func(*args, **kwargs)
+            return otherwise
+        return wrapper
+    return decorator
+
+
+@enterprise_is_enabled()
+def get_enterprise_customer_cache_key(uuid, username=settings.ENTERPRISE_SERVICE_WORKER_USERNAME):
+    """The cache key used to get cached Enterprise Customer data."""
+    return get_cache_key(
+        resource='enterprise-customer',
+        resource_id=uuid,
+        username=username,
+    )
+
+
+@enterprise_is_enabled()
+def cache_enterprise(customer):
+    """Cache this customer's data."""
+    cache_key = get_enterprise_customer_cache_key(customer['uuid'])
+    cache.set(cache_key, customer, settings.ENTERPRISE_API_CACHE_TIMEOUT)
```

Comment:
@UmanShahzad Nit: It might be worth sticking with `enterprise_customer` terminology here (instead of the plain `customer`).

Reply:
:+1: